main.tf

```diff
@@ -51,223 +54,66 @@ locals {
-      0,
-    )
-  }
-
-  cluster_type_output_region = {
-    regional = element(concat(google_container_cluster.primary.*.region, [""]), 0)
-    zonal    = var.region
-  }
-
-  cluster_type_output_regional_zones = flatten(google_container_cluster.primary.*.node_locations)
-  cluster_type_output_zonal_zones    = local.zone_count > 1 ? slice(var.zones, 1, local.zone_count) : []
-
-  cluster_type_output_zones = {
-    regional = local.cluster_type_output_regional_zones
-    zonal = concat(
-      google_container_cluster.zonal_primary.*.zone,
-      local.cluster_type_output_zonal_zones,
-    )
-  }
-
-  cluster_type_output_endpoint = {
-    regional = element(concat(google_container_cluster.primary.*.endpoint, [""]), 0)
-    zonal = element(
-      concat(google_container_cluster.zonal_primary.*.endpoint, [""]),
-      0,
-    )
-  }
-
-  cluster_type_output_master_auth = {
-    regional = concat(google_container_cluster.primary.*.master_auth, [])
-    zonal    = concat(google_container_cluster.zonal_primary.*.master_auth, [])
-  }
-
-  cluster_type_output_master_version = {
-    regional = element(
-      concat(google_container_cluster.primary.*.master_version, [""]),
-      0,
-    )
-    zonal = element(
-      concat(
-        google_container_cluster.zonal_primary.*.master_version,
-        [""],
-      ),
-      0,
-    )
-  }
 
-  cluster_type_output_min_master_version = {
-    regional = element(
-      concat(google_container_cluster.primary.*.min_master_version, [""]),
-      0,
-    )
-    zonal = element(
-      concat(
-        google_container_cluster.zonal_primary.*.min_master_version,
-        [""],
-      ),
-      0,
-    )
-  }
+  cluster_output_name           = google_container_cluster.primary.name
+  cluster_output_location       = google_container_cluster.primary.location
+  cluster_output_region         = google_container_cluster.primary.region
+  cluster_output_regional_zones = google_container_cluster.primary.node_locations
+  cluster_output_zonal_zones    = local.zone_count > 1 ? slice(var.zones, 1, local.zone_count) : []
+  cluster_output_zones          = local.cluster_output_regional_zones
 
-  cluster_type_output_logging_service = {
-    regional = element(
-      concat(google_container_cluster.primary.*.logging_service, [""]),
-      0,
-    )
-    zonal = element(
-      concat(
-        google_container_cluster.zonal_primary.*.logging_service,
-        [""],
-      ),
-      0,
-    )
-  }
+  cluster_output_endpoint = google_container_cluster.primary.endpoint
 
-  cluster_type_output_monitoring_service = {
-    regional = element(
-      concat(google_container_cluster.primary.*.monitoring_service, [""]),
-      0,
-    )
-    zonal = element(
-      concat(
-        google_container_cluster.zonal_primary.*.monitoring_service,
-        [""],
-      ),
-      0,
-    )
-  }
+  cluster_output_master_auth                        = concat(google_container_cluster.primary.*.master_auth, [])
```

Comment:
@morgante  usage of cluster_output_master_auth looks suspicious to me  https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/0002cd7811daef9eede83ab39da5322bda0d9145/main.tf#L81-L83

Reply:
Frankly, I think we can/should drop this output entirely. It's likely not used, nor important. Just add a note to the CHANGELOG that it was removed.

Reply:
only part is used here is cluster_master_auth_map, only to get certificate, 
`cluster_ca_certificate                     = local.cluster_master_auth_map["cluster_ca_certificate"]`
but the way cluster_master_auth_map variable retrieved looks strange, I'll check for more elegant way

Reply:
for now  I would prefer to leave code as it is, as long it works stable